pkg/apis/meta/v1/time.go

```diff
@@ -83,6 +83,16 @@ func (t Time) Equal(u Time) bool {
 	return t.Time.Equal(u.Time)
 }
 
+// BeforeMicro reports whether the time instant t is before micro instance u.
```

Comment:
I don't think we should do that. Unversioned Time is used internally as normal timestamp with nanosecond precision, the only thing that I'm not happy with is that when serialized it truncates time to second. The second 'MicroTime' is designed to be exactly the same, but serialized differently. Hence actually we should compare 'full' representations.